File: framework/wcf/Scenario/Federation/CS/HomeRealmSTS/HomeRealmSTSHost.cs
//-----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//-----------------------------------------------------------------------------
using System;

using System.ServiceModel;
using System.ServiceModel.Activation;

namespace Microsoft.Samples.Federation
{
	public class HomeRealmSTSHostFactory : ServiceHostFactoryBase
	{
	    public override ServiceHostBase CreateServiceHost(string constructorString, Uri[] baseAddresses)
	    {
	        return new HomeRealmSTSHost(baseAddresses);
	    }
	}
	
	class HomeRealmSTSHost : ServiceHost
	{
        public HomeRealmSTSHost(params Uri[] addresses) : base(typeof(HomeRealmSTS), addresses)
		{
            ServiceConstants.LoadAppSettings();
		}
	}
}

File: framework/wcf/Scenario/Federation/VB/HomeRealmSTS/HomeRealmSTSHost.vb
' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System

Imports System.ServiceModel
Imports System.ServiceModel.Activation

Namespace Microsoft.Samples.Federation

    Public Class HomeRealmSTSHostFactory
        Inherits ServiceHostFactoryBase

        Public Overloads Overrides Function CreateServiceHost(ByVal constructorString As String, ByVal baseAddresses() As Uri) As ServiceHostBase

            Return New HomeRealmSTSHost(baseAddresses)

        End Function

    End Class

    Class HomeRealmSTSHost
        Inherits ServiceHost

        Public Sub New(ByVal ParamArray addresses() As Uri)

            MyBase.New(GetType(HomeRealmSTS), addresses)
            ServiceConstants.LoadAppSettings()

        End Sub

    End Class

End Namespace

